executor/plan_replayer.go

```diff
@@ -146,16 +155,56 @@ func (e *PlanReplayerExec) Next(ctx context.Context, req *chunk.Chunk) error {
 	if e.endFlag {
 		return nil
 	}
-	if e.ExecStmts == nil {
+	err := e.createFile(domain.GetPlanReplayerDirName())
+	if err != nil {
+		return err
+	}
+	if len(e.DumpInfo.Path) > 0 {
+		err = e.prepare()
+		if err != nil {
+			return err
+		}
+		e.endFlag = true
+		e.ctx.GetSessionVars().LastPlanReplayerToken = e.DumpInfo.FileName
+		return nil
+	}
+	if e.DumpInfo.ExecStmts == nil {
 		return errors.New("plan replayer: sql is empty")
 	}
-	res, err := e.dump(ctx, domain.GetPlanReplayerDirName())
+	err = e.DumpInfo.dump(ctx)
 	if err != nil {
 		return err
 	}
-	req.AppendString(0, res)
+	req.AppendString(0, e.DumpInfo.FileName)
 	e.endFlag = true
-	e.ctx.GetSessionVars().LastPlanReplayerToken = res
+	e.ctx.GetSessionVars().LastPlanReplayerToken = e.DumpInfo.FileName
+	return nil
+}
+
+func (e *PlanReplayerExec) createFile(path string) error {
+	// Create path
+	err := os.MkdirAll(path, os.ModePerm)
+	if err != nil {
+		return errors.AddStack(err)
+	}
+
+	// Generate key and create zip file
+	time := time.Now().UnixNano()
+	b := make([]byte, 16)
+	//nolint: gosec
+	_, err = rand.Read(b)
+	if err != nil {
+		return err
+	}
+	key := base64.URLEncoding.EncodeToString(b)
+	fileName := fmt.Sprintf("replayer_%v_%v.zip", key, time)
+	zf, err := os.Create(filepath.Join(path, fileName))
+	if err != nil {
+		return errors.AddStack(err)
+	}
+	e.DumpInfo.File = zf
+	e.DumpInfo.FileName = fileName
+	e.ctx.GetSessionVars().LastPlanReplayerToken = e.DumpInfo.FileName
```

Comment:
If some error happens when dumping info, do we need to clear `LastPlanReplayerToken`?

Reply:
updated.